baseapp/queryrouter.go

```diff
@@ -0,0 +1,51 @@
+package baseapp
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// QueryRouter provides queryables for each query path.
+type QueryRouter interface {
+	AddRoute(r string, h sdk.Querier) (rtr QueryRouter)
+	Route(path string) (h sdk.Querier)
+}
+
+// map a transaction type to a handler and an initgenesis function
+type queryroute struct {
+	r string
+	h sdk.Querier
+}
+
+type queryrouter struct {
+	routes []queryroute
```

Comment:
why not make it a `map[string]queryroute`?

Reply:
Idk, this is the way the normal router is, and so I just copied that.  I think it may have had to do with allowing for more complex routes in the future (like overriding "gov/proposal" to go somewhere different thatn "gov/*").  @jaekwon

Reply:
Mmm I'd be fine w/ either map[string]queryroute or not, I think QueryRouter might/should be simpler.  But either way duplicate registration should panic.

Reply:
Ok, switched